SophonPatchAsset.Update.cs

```diff
@@ -0,0 +1,482 @@
+                    targetFileStream,
+                    checkByHashChunk.ChunkHashDecompressed,
+                    true,
+                    token) :
+                await checkByHashChunk.CheckChunkMd5HashAsync(targetFileStream,
+                    true,
+                    token);
+
+            return isHashMatched;
+        }
+    }
+
+    internal class PatchTargetProperty : IDisposable
+    {
+        private FileInfo TargetFileInfo { get; }
+        public FileInfo TargetFileTempInfo { get; }
+        public FileStream? TargetFileTempStream { get; }
+        public string PatchFilePath { get; }
+        private FileStream? PatchFileStream { get; }
+        public ChunkStream? PatchChunkStream { get; }
+
+        private PatchTargetProperty(string patchOutputDir, string patchNameSource, string inputDir, string targetFilePath, long patchOffset, long patchLength, bool createStream)
+        {
+            PatchFilePath = Path.Combine(patchOutputDir, patchNameSource);
+            targetFilePath = Path.Combine(inputDir, targetFilePath);
+            string targetFileTempPath = targetFilePath + ".temp";
+
+            TargetFileInfo = new FileInfo(targetFilePath);
+            TargetFileTempInfo = new FileInfo(targetFileTempPath);
+            TargetFileTempInfo.Directory?.Create();
+
+            if (TargetFileTempInfo.Exists)
+            {
+                TargetFileTempInfo.IsReadOnly = false;
+                TargetFileTempInfo.Refresh();
+            }
+
+            if (!File.Exists(PatchFilePath))
+            {
+                throw new FileNotFoundException($"Required patch file: {PatchFilePath} is not found!");
+            }
+
+            if (!createStream)
+            {
+                return;
+            }
+
+            long patchChunkEnd = patchOffset + patchLength;
+            TargetFileTempStream = TargetFileTempInfo.Open(FileMode.Create, FileAccess.Write, FileShare.Write);
+            PatchFileStream = File.Open(PatchFilePath, FileMode.Open, FileAccess.Read, FileShare.Read);
+            PatchChunkStream = new ChunkStream(PatchFileStream, patchOffset, patchChunkEnd);
+        }
+
+        public static PatchTargetProperty Create(string patchOutputDir, string patchNameSource, string inputDir, string targetFilePath, long patchOffset, long patchLength, bool createTempStream)
+            => new(patchOutputDir, patchNameSource, inputDir, targetFilePath, patchOffset, patchLength, createTempStream);
+
+        public void Dispose()
+        {
+            PatchChunkStream?.Dispose();
+            PatchFileStream?.Dispose();
+            TargetFileTempStream?.Dispose();
+
+            TargetFileTempInfo.Refresh();
+            if (TargetFileInfo.Exists)
+            {
+                TargetFileInfo.IsReadOnly = false;
+                TargetFileInfo.Delete();
+            }
+
+            TargetFileTempInfo.MoveTo(TargetFileInfo.FullName);
```

Comment:
Instead of doing this in Dispose method, I think we can move is wherever it needs to be called. Or if you prefer to have it here for deduplication purpose, I think we can move them to before any Dispose calls

Reply:
Unfortunately, the MoveTo will throw SharingViolation error before it's getting disposed. This is the reason why I put it at the very end of every calls